### PR TITLE
Refresh travis

### PR DESCRIPTION
**Pull request by mrakitin**
This PR is to make `.travis.yml` up to date.

I removed `circle.yml as it was quite outdated and was not triggering builds.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,65 +1,47 @@
...
 
 cache:
   directories:
     - $HOME/.cache/pip
 
-matrix:
+jobs:
   include:
     - python: 3.6
+    - python: 3.7
```
Comment (prjemian) [inline .travis.yml:12]: add py38, too?

Comment (mrakitin) [reply .travis.yml:12]: Good idea. We don't have all packages in the `nsls2forge` channel for Py38 yet, but there are most of them.

File: .travis.yml [hunk 2]
```diff
@@ -1,65 +1,47 @@
...
+
+  - source /home/travis/mc/etc/profile.d/conda.sh
 
-  - export PATH=/home/travis/mc/bin:$PATH
   - conda config --set always_yes true
   - conda update conda --yes
-  - conda install conda-build anaconda-client jinja2
-  - conda config --add channels lightsource2-tag
-  - conda config --add channels soft-matter
+  - conda config --add channels nsls2forge
```
Comment (prjemian) [inline .travis.yml:27]: Is it possible to put all this into an [`environment.yml`](https://github.com/BCDA-APS/apstools/blob/master/environment.yml) file (channels, packages, pip installs)?

File: .travis.yml [hunk 3]
```diff
@@ -1,65 +1,47 @@
...
-  - source activate $CONDA_ENV
+  - conda deactivate
+  - conda activate $CONDA_ENV
   - conda list
   - py.test -vv --cov=hkl --cov-report term-missing hkl/tests
 
 after_success:
-    - env
-    - coveralls
+  - env | sort
```
Comment (prjemian) [inline .travis.yml:46]: 👍

File: .travis.yml (hunk 1 above)
Comment (mrakitin) [reply .travis.yml:12]: Done.

File: .travis.yml (hunk 2 above)
Comment (prjemian) [reply .travis.yml]: Looking good.

File: environment.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+name: hklpy
```
Comment (prjemian) [inline environment.yml:1]: error: `CondaValueError: could not parse 'name: hklpy' in: environment.yml`

change to `hklpyenv` (just avoid `hklpy` string)

Comment (mrakitin) [reply environment.yml:1]: I think the problem is that `conda create -n ...` and `conda env create ...` accept files with different formats. While it seemed more convenient to use `conda create -n ...` with the desired Python version in one shot, it does not seem to work. I went your way - replacing Python version with exact pinning.

**Review by prjemian — APPROVED** [review]
LGTM

**Comment by prjemian** [comment]
Nice job.  Thanks for taking the lead!

**Comment by mrakitin** [comment]
Thanks for reviewing and useful suggestions, @prjemian!

**Pull request merged**